### PR TITLE
[Runtime] Refactor HeapObject override checks to use a macro.

### PR DESCRIPTION
**Pull request by mikeash**
This gives us a one-stop shop for making changes to the override mechanism.

rdar://problem/54752086

### CODE REVIEW

**Comment by mikeash** [comment]
@swift-ci please test and merge

**Comment by mikeash** [comment]
@swift-ci please test linux platform

File: stdlib/public/runtime/HeapObject.cpp [hunk 1]
```diff
@@ -72,6 +72,19 @@ static inline bool isValidPointerForNativeRetain(const void *p) {
 #endif
 }
 
+// Call the appropriate implementation of the `name` function, passing `args`
+// to the call. This checks for an override in the function pointer. If an
+// override is present, it calls that override. Otherwise it directly calls
+// the default implementation. This allows the compiler to inline the default
+// implementation and avoid the penformance penalty of indirecting through
```
Comment (jrose-apple) [inline stdlib/public/runtime/HeapObject.cpp]: Typo: "penformance"

Comment (mikeash) [reply stdlib/public/runtime/HeapObject.cpp]: Argh. You're not even the first one to point this out. I'm just bad at fixing it.

**Comment by mikeash** [comment]
@swift-ci please test

**Comment by mikeash** [comment]
@swift-ci please test linux platform

**Pull request merged**